### PR TITLE
Recognize versions from tags, not only bower.json

### PR DESCRIPTION
**Pull request by sheerun**
For example for moment#0.3.2. That version exists in tags, but has no bower.json. If yoy as `bower info momentjs` we clearly see that this version exists, so we schedule build, but during actual build, no version is defined in bower.json, so it fails.

One solution is to fetch all component information from bower, and leave bower.json/package.json for it to parse.
- [x] Use bower info for fetching repository description, main files and dependencies
- [x] Use bower install instead of bower info to get component information + extract information about dependencies
- [x] Integrate Converter class with the rest of application of test manually if everithing works
- [x] Only one bower install per request

### CODE REVIEW

**Comment by sheerun** [comment]
I'm doing it.

**Comment by sheerun** [comment]
@teamon I won't be touching this branch for now, so you can continue. Just commit early.

**Comment by sheerun** [comment]
@teamon For some reason jquery-waypoints don't have main file now, so tests changed. Also resizeend includes only coffee files (they have priority with new implementation if main file is missing).

File: app/models/build/bower_component.rb [hunk 1]
```diff
@@ -22,7 +26,7 @@ def github!(user)
       self.user = user
 
       self.repository = "git://github.com/#{user}/#{name}.git"
-      self.homepage = self.class.get_homepage_from_repository(repository)
+      self.homepage = "http://github.com/#{user}/#{name}"
```
Comment (teamon) [inline app/models/build/bower_component.rb]: What about other possible homepages?

File: app/models/build/bower_component.rb [hunk 2]
```diff
@@ -53,43 +57,17 @@ def gem
...
-          version:      data["version"],
-          description:  data["description"],
-          main:         data["main"] ? [data["main"]].flatten.reject {|e| e.nil?} : nil,
-          dependencies: data["dependencies"],
-          repository:   data["repository"].is_a?(Hash) ? data["repository"]["url"] : data["repository"],
-          homepage:     data["homepage"]
-        }.reject {|k,v| v.blank? }
+    def self.from_bower(name, version = nil)
+      full_name = version ?  name + "##{version}" : name
+      json = Utils.bower('/tmp', "info #{full_name}")
```
Comment (teamon) [inline app/models/build/bower_component.rb]: I know `bower info` does not touch FS but I still don't like the bare `/tmp` here

File: config/initializers/sidekiq.rb [hunk 3]
```diff
@@ -1,7 +1,13 @@
...
+    :namespace => REDIS_NAMESPACE
+  }
 end
 
 Sidekiq.configure_client do |config|
-  config.redis = { :namespace => REDIS_NAMESPACE }
+  config.redis = {
+    :url => ENV['REDIS_URL'],
+    :namespace => REDIS_NAMESPACE
+  }
```
Comment (teamon) [inline config/initializers/sidekiq.rb:5]: Doesn't sidekiq catch `REDIS_URL` by default (if exists)?

File: app/models/build/bower_component.rb (hunk 1 above)
Comment (sheerun) [inline app/models/build/bower_component.rb]: Bower always set such homepage.

File: app/models/build/bower_component.rb (hunk 2 above)
Comment (sheerun) [inline app/models/build/bower_component.rb]: What path would you use? Btw. it's your solution in different place.

File: config/initializers/sidekiq.rb (hunk 3 above)
Comment (sheerun) [inline config/initializers/sidekiq.rb:5]: No idea.

**Pull request merged**